File: delivery-service/api/src/main/java/org/delivery/api/common/api/Result.java
package org.delivery.common.api;

/*@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Result {

    private Integer resultCode;
    private String resultMessage;
    private String resultDescription;

    public static Result ok(){
        return Result.builder()
                .resultCode(ErrorCode.OK.getErrorCode())
                .resultMessage(ErrorCode.OK.getDescription())
                .resultDescription("성공")
                .build();
    }

    public static Result ERROR(errorCodeIfs errorCodeIfs, Throwable tw){
        return Result.builder()
                .resultCode(errorCodeIfs.getErrorCode())
                .resultMessage(errorCodeIfs.getDescription())
                .resultDescription(tw.getLocalizedMessage())
                .build();
    }

    public static Result ERROR(errorCodeIfs errorCodeIfs, String description){
        return Result.builder()
                .resultCode(errorCodeIfs.getErrorCode())
                .resultMessage(errorCodeIfs.getDescription())
                .resultDescription(description)
                .build();
    }

    public static Result ERROR(errorCodeIfs errorCodeIfs) {
        return Result.builder()
                .resultCode(errorCodeIfs.getErrorCode())
                .resultMessage(errorCodeIfs.getDescription())
                .resultDescription("성공")
                .build();
    }
}*/

File: delivery-service/common/src/main/kotlin/org/delivery/common/api/Result.kt
package org.delivery.common.api

import org.delivery.common.error.ErrorCode
import org.delivery.common.error.errorCodeIfs

data class Result (
    val resultCode: Int?= null,
    val resultMessage: String?= null,
    val resultDescription: String?= null
){
    //java의 static
    companion object {
        @JvmStatic
        fun ok(): Result {
            return Result(
                resultCode = ErrorCode.OK.getErrorCode(),
                resultMessage = ErrorCode.OK.getDescription(),
                resultDescription = "성공"
            )
        }

        @JvmStatic
        fun ERROR(errorCodeIfs: errorCodeIfs): Result{
            return Result(
                resultCode = errorCodeIfs.getErrorCode(),
                resultMessage = errorCodeIfs.getDescription(),
                resultDescription = "에러발생"
            )
        }

        @JvmStatic
        fun ERROR(errorCodeIfs: errorCodeIfs, tw:Throwable): Result{
            return Result(
                resultCode = errorCodeIfs.getErrorCode(),
                resultMessage = errorCodeIfs.getDescription(),
                resultDescription = tw.localizedMessage
            )
        }

        @JvmStatic
        fun ERROR(errorCodeIfs: errorCodeIfs, description: String): Result{
            return Result(
                resultCode = errorCodeIfs.getErrorCode(),
                resultMessage = errorCodeIfs.getDescription(),
                resultDescription = description
            )
        }
    }
}